### PR TITLE
[CI:DOCS] podman-generate-systemd.1.md: document --sdnotify

### LINKED ISSUE #15029: --sdnotify default value from the "podman create" man page is "container" but "conmon" is used when running "podman generate systemd ..."
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

__--sdnotify__ has the default value _container_ according to the __podman create__ man page but the value _conmon_ is used instead when running `podman generate systemd ...`

https://github.com/containers/podman/blob/04ed519e9d38e2ae12ec0b568e6418e750ff203d/docs/source/markdown/podman-create.1.md?plain=1#L948

**Steps to reproduce the issue:**

1. Run the commands
    ```
    $ podman create --rm --name test1 ghcr.io/eriksjolund/socket-activate-echo
    4ec5ea126989c6926ed9ce05fcdf713209413b2dd083984c5ea40c80e15d4877
    $ podman generate systemd --name --new test1 | grep sdnotify
    	--sdnotify=conmon \
    $ 
    ```

**Describe the results you received:**

```
	--sdnotify=conmon \
```

**Describe the results you expected:**

If _container_ is the default value, I would expect the result to be

```
	--sdnotify=container \
```

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
Client:       Podman Engine
Version:      4.1.1
API Version:  4.1.1
Go Version:   go1.18.3
Built:        Wed Jun 22 18:17:44 2022
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.26.1
  cgroupControllers:
  - cpu
  - io
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.0-2.fc36.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.0, commit: '
  cpuUtilization:
    idlePercent: 99.91
    systemPercent: 0.03
    userPercent: 0.05
  cpus: 16
  distribution:
    distribution: fedora
    variant: workstation
    version: "36"
  eventLogger: journald
  hostname: asus
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1071
      size: 1
    - container_id: 1
      host_id: 4511264
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1071
      size: 1
    - container_id: 1
      host_id: 4511264
      size: 65536
  kernel: 5.18.11-200.fc36.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 2152271872
  memTotal: 7578664960
  networkBackend: netavark
  ociRuntime:
    name: crun
    package: crun-1.4.5-1.fc36.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.4.5
      commit: c381048530aa750495cf502ddb7181f2ded5b400
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    path: /run/user/1071/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-0.2.beta.0.fc36.x86_64
    version: |-
      slirp4netns version 1.2.0-beta.0
      commit: 477db14a24ff1a3de3a705e51ca2c4c1fe3dda64
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.3
  swapFree: 7574908928
  swapTotal: 7578054656
  uptime: 31h 46m 39.4s (Approximately 1.29 days)
plugins:
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /home/test159/.config/containers/storage.conf
  containerStore:
    number: 9
    paused: 0
    running: 1
    stopped: 8
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/test159/.local/share/containers/storage
  graphRootAllocated: 407822663680
  graphRootUsed: 350593806336
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 12
  runRoot: /run/user/1071/containers
  volumePath: /home/test159/.local/share/containers/storage/volumes
version:
  APIVersion: 4.1.1
  Built: 1655914664
  BuiltTime: Wed Jun 22 18:17:44 2022
  GitCommit: ""
  GoVersion: go1.18.3
  Os: linux
  OsArch: linux/amd64
  Version: 4.1.1

```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-4.1.1-2.fc36.x86_64
```

### PR DESCRIPTION
**Pull request by eriksjolund**
Fixes #15029

Signed-off-by: Erik Sjölund <erik.sjolund@gmail.com>

#### Does this PR introduce a user-facing change?

```release-note
None
```

This PR clarifies things a bit. (It could probably be improved)

### CODE REVIEW

File: docs/source/markdown/podman-generate-systemd.1.md [hunk 1]
```diff
@@ -14,6 +14,8 @@ Generating unit files for a pod requires the pod to be created with an infra con
 
 _Note: If you use this command with the remote client, including Mac and Windows (excluding WSL2) machines, you would still have to place the generated units on the remote system.  Moreover, please make sure that the XDG_RUNTIME_DIR environment variable is set.  If unset, you may set it via `export XDG_RUNTIME_DIR=/run/user/$(id -u)`._
 
+_Note: The generated `podman run` command contains an `--sdnotify` option with the value taken from the container, unless the container value was set to __ignore__, then the value __conmon__ is used. If the container does not have any explicitly set value, the value __conmon__ is used._
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-generate-systemd.1.md:17]:
```suggestion
_Note: The generated `podman run` command contains an `--sdnotify` option with the value taken from the container. If the container does not have any explicitly set value or the value is set to __ignore__, the value __conmon__ is used._
```

**Comment by eriksjolund** [comment]
I made this a draft as there are more things to fix.


I noticed that
https://github.com/containers/podman/blob/main/docs/source/markdown/podman-generate-systemd.1.md#generate-systemd-unit-file-for-a-container-with---new-flag
has an old example where the generated output contains

```
Type=forking
```
Recent Podman versions generate 
```
Type=notify
```

**Comment by eriksjolund** [comment]
I'll do a separate PR for fixing the very old examples. Let's keep this PR specific.

I incorporated some text from the [comment](https://github.com/containers/podman/issues/15029#issuecomment-1192244755) by @vrothberg.

Because of that, I added 
```
 Co-authored-by: Valentin Rothberg <vrothberg@redhat.com>
```
to the commit message. (I'm not sure what the correct procedure is)

**Review by vrothberg — COMMENTED** [review]
/lgtm hold

@Luap99 mind having a final look?

**Comment by vrothberg** [comment]
Thanks, @eriksjolund !

**Pull request merged**